### PR TITLE
#68 - SliceIndex implementation corrected and tested

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #68 I've corrected `SliceIndex` implementation and added tests for it, also corrected `PySlice` and used it in `PypiCliITCase` to be sure pip understands our index.

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/pypi-adapter/issues/70#issuecomment-658071556))

@olenagerasimova Job `gh:artipie/pypi-adapter#70` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/pypi-adapter/issues/70#issuecomment-658071556))

@olenagerasimova Pull request [#70](https://github.com/artipie/pypi-adapter/issues/70) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

File: src/test/java/com/artipie/pypi/http/SliceIndexTest.java [hunk 1]
```diff
@@ -27,38 +27,142 @@
...
-                // @checkstyle LineLengthCheck (1 line)
-                "<!DOCTYPE html>\n<html>\n  </body>\n<a href=\"/abc\">abc</a><br/>\n</body>\n</html>".getBytes()
+                Matchers.anyOf(
+                    new IsEqual<>(SliceIndexTest.html(two, one, three)),
+                    new IsEqual<>(SliceIndexTest.html(two, three, one)),
+                    new IsEqual<>(SliceIndexTest.html(three, one, two)),
+                    new IsEqual<>(SliceIndexTest.html(three, two, one)),
+                    new IsEqual<>(SliceIndexTest.html(one, three, two)),
+                    new IsEqual<>(SliceIndexTest.html(one, two, three))
+                )
```
Comment (olegmoz) [inline src/test/java/com/artipie/pypi/http/SliceIndexTest.java:118]: @olenagerasimova this kind of code for matching ignoring order is duplicated in tests. What do you think about extracting it to a method? You can use something like this: https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Collections2.html#permutations(java.util.Collection) to build permutations list.

Also it is not clear to me why you want to allow any order - I believe `Storage.list()` method produces ordered list of keys.

Comment (olenagerasimova) [reply src/test/java/com/artipie/pypi/http/SliceIndexTest.java]: @olegmoz thanks, I've added this information to `Storage` javadoc https://github.com/artipie/asto/pull/211, looks like all our implementations return sorted lists and get rid of permutations here.

**Comment by 0crat** [comment]
Job `gh:artipie/pypi-adapter#70` is not assigned, can't get performer

**Pull request merged**